llumnix/backends/vllm/worker.py

```diff
@@ -121,8 +139,47 @@ def migrate_cache(self, src_worker_handle_list, src_blocks: List[int], dst_block
     def do_recv(self, *args, **kwargs):
         return self.migration_backend.do_recv(*args, **kwargs)
 
-    def do_send(self, *args, **kwargs):
-        return self.migration_backend.do_send(*args, **kwargs)
+    def do_send(self, *args, request_id: str = None, send_worker_data: bool = False, **kwargs):
+        if not send_worker_data:
+            return self.migration_backend.do_send(*args, **kwargs)
+        return self.migration_backend.do_send(*args, **kwargs), self.get_data_src(request_id)
```

Comment:
I think the term "data" looks somewhat ambiguous to me in this PR, like in "worker data", "get data src", etc. Could you please use some term more specific, maybe request metadata or something like that?

Reply:
Also, I'd prefer renaming `get_data_src` to a name more similar to `do_send`, like `do_send_metadata`. Because you call `do_send` and `get_data_src` in a `do_send` looks confusing.

Reply:
The data here represents the `_seq_group_metadata_cache` of worker. More accurately, `get_data_src` can be renamed to `get_seq_group_metadata_src`. `do_send_metadata` is not appropriate because this function is used to get the metadata of sequence group stored in worker. And `do_send` is used to trigger src migration backend to do send communication.